native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe/deserializer/GlueSchemaRegistryKafkaDeserializer.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections.Generic;
+using AWSGsrSerDe.common;
+
+namespace AWSGsrSerDe.deserializer
+{
+    public class GlueSchemaRegistryKafkaDeserializer
+    {
+        private readonly DeserializerFactory _deserializerFactory = DeserializerFactory.GetInstance();
+        private readonly GlueSchemaRegistryConfiguration _configuration;
+
+        public GlueSchemaRegistryKafkaDeserializer(Dictionary<string, dynamic> configs)
+        {
+            _configuration = new GlueSchemaRegistryConfiguration(configs);
+        }
+        
+        public object Deserialize(string topic, byte[] data)
+        {
+            if (null == data)
+            {
+                return null;
+            }
+
+            var glueSchemaRegistryDeserializer = new GlueSchemaRegistryDeserializer();
```

Comment:
This is a very expensive call and creates a new instance of underlying cache preventing successive calls from caching schemas. We need to maintain one instance of this per KafkaDeserializer.

Reply:
nice catch, I will update that.

Reply:
Not updated?

Reply:
sorry about missing it, I just updated it.